### PR TITLE
Add support for Attach API functionality

### PR DESCRIPTION
**Pull request by aka-somix**
Extended virtual kubelet API with the "Attach" subresource.

### Added
- Attach funtionality to remote commands
- Attach functionality API handler

### Changed
- update PodHandlerConfig to also manage attach method
- update Provider interface to also have an AttachToContainer property

### CODE REVIEW

**Comment by aka-somix** [comment]
Hi! 
Thank you. I have rebased my branch to align it with the master branch, but pushing it caused the Unit tests check to fail and it do not know if I can run again checks.
I saw that unit tests basically timed out, there is something we can do to run them again or you are going to do that for us?

File: node/api/attach.go [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+	"github.com/virtual-kubelet/virtual-kubelet/internal/kubernetes/remotecommand"
+	"k8s.io/apimachinery/pkg/types"
+	remoteutils "k8s.io/client-go/tools/remotecommand"
+)
+
+// ContainerExecHandlerFunc defines the handler function used for "execing" into a
+// container in a pod.
+type ContainerAttachHandlerFunc func(ctx context.Context, namespace, podName, containerName string, attach AttachIO) error
+
+// HandleContainerExec makes an http handler func from a Provider which execs a command in a pod's container
```
Comment (cpuguy83) [inline node/api/attach.go:35]: doc string is for exec instead of attach

File: node/nodeutil/provider.go [hunk 2]
```diff
@@ -28,6 +28,10 @@ type Provider interface {
 	// between in/out/err and the container's stdin/stdout/stderr.
 	RunInContainer(ctx context.Context, namespace, podName, containerName string, cmd []string, attach api.AttachIO) error
 
+	// AttachToContainer attaches to the executing process of a container in the pod, copying data
+	// between in/out/err and the container's stdin/stdout/stderr.
+	AttachToContainer(ctx context.Context, namespace, podName, containerName string, attach api.AttachIO) error
```
Comment (cpuguy83) [inline node/nodeutil/provider.go:33]: Let's make sure to call this change out in the release notes

**Review by cpuguy83 — CHANGES_REQUESTED** [review]

File: node/api/attach.go [hunk 3]
```diff
@@ -28,11 +28,11 @@ import (
 	remoteutils "k8s.io/client-go/tools/remotecommand"
 )
 
-// ContainerExecHandlerFunc defines the handler function used for "execing" into a
+// ContainerAttachHandlerFunc defines the handler function used for "execing" into a
```
Comment (pires) [inline node/api/attach.go:31]: The linter didn't complain (?) :(

**Pull request merged**